api/tests/Behat/bootstrap/v2/Common/ExpectedResultsTrait.php

```diff
@@ -155,4 +185,72 @@ private function raiseExpectedResultsException($expectedItems, $foundItems, $sec
             $this->getCurrentUrl()
         );
     }
+
+    private function debugExpectedResultsDisplayed($sections, $summarySectionNumber, $expectedResults)
+    {
+        $summarySectionsText = '';
+        foreach ($sections as $sectionKey => $section) {
+            $summarySectionsText = $summarySectionsText."\n\nSection Number: ".strval($sectionKey)."\n";
+            foreach ($section as $rowNumber => $row) {
+                $summarySectionsText = $summarySectionsText."\tRow Number: ".strval($rowNumber)."\n";
+                foreach ($row as $fieldNumber => $field) {
+                    $summarySectionsText = $summarySectionsText."\t\t".strtolower(strval($field))."\n";
+                }
+            }
+        }
+
+        $expectedText = "\n\nThe input to this function is specifically looking at section: ".strval($summarySectionNumber)."\n";
+        $expectedText = $expectedText."\n\nSection Number: ".strval($summarySectionNumber)."\n";
+        foreach ($expectedResults as $rowNumber => $row) {
+            $expectedText = $expectedText."\tRow Number: ".strval($rowNumber)."\n";
+            foreach ($row as $fieldNumber => $field) {
+                $expectedText = $expectedText."\t\t".strtolower(strval($field))."\n";
+            }
+        }
+
+        $message = <<<MESSAGE
+
+============================
+-- DEBUG REPORT --
+
+How to use: Firstly, check that the elements under 'found' correspond with what you see on
+the summary screen.
+
+If there are missing items or sections on the screen that don't appear in the found section
+then you may have found an edge case that won't work with this function.
+Either amend the function or use some bespoke code.
+
+Secondly, check that the items you are expecting appear in the equivalent 'Section Number'
+and 'Row Number' and that the 'fields' appear in the right order.
+
+It is completely fine to have more rows in the 'found' section. As long as they are in the
+correct section, row and field order.
+
+-- FOUND --
+%s
+
+-- EXPECTED --
+%s
+
+============================
+
+MESSAGE;
+
+        $this->throwContextualException(sprintf(
+            $message,
+            $summarySectionsText,
+            $expectedText
+        ));
+    }
+
+    private function checkExpectedResultsCorrectFormat($expectedResults)
+    {
+        foreach ($expectedResults as $expectedResult) {
+            if (!is_array($expectedResult)) {
+                $this->throwContextualException(
+                    "\nIncorrect data types - \nexpectedResults must be an array of array of strings to represent rows and fields on summary page"
+                );
+            }
```

Comment:
Depends on how far we want to go but you could also check all of the elements are strings here too as the error message makes it sounds like we are already checking this. Another `foreach` and `is_string()` would do it.

Reply:
yep done